fapolicy_analyzer/ui/rules_difference_dialog.py

```diff
@@ -0,0 +1,68 @@
+# Copyright Concurrent Technologies Corporation 2021
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <https://www.gnu.org/licenses/>.
+import gi
+
+from fapolicy_analyzer import System, rules_difference
+from fapolicy_analyzer.ui.ui_widget import UIBuilderWidget
+from fapolicy_analyzer.ui.searchable_list import SearchableList
+
+gi.require_version("Gtk", "3.0")
+from gi.repository import Gtk  # isort: skip
+
+
+class RulesDifferenceDialog(UIBuilderWidget):
+    def __init__(self,
+                 current_system: System,
+                 previous_system: System,
+                 parent=None,
+                 ):
+        super().__init__()
+        if parent:
+            self.get_ref().set_transient_for(parent)
+
+        diffs = rules_difference(previous_system, current_system).split("\n")
+        new_list = SearchableList(self.__columns())
+        prev_list = SearchableList(self.__columns())
+        self.get_object("newContent").add(new_list.get_ref())
+        self.get_object("prevContent").add(prev_list.get_ref())
+        new_store = Gtk.ListStore(str)
+        prev_store = Gtk.ListStore(str)
+
+        for d in diffs:
+            if (
+               (d.startswith("+") and d.split("+")[-1] + "\n" in previous_system.rules_text())
+               or (d.startswith("-") and d.split("-")[-1].split("\n")[0] in current_system.rules_text())
+               ):
+                continue
+
+            if d.startswith("+") and d.split("+")[-1] in current_system.rules_text():
+                new_store.append([d])
+            elif d.startswith("-") and d.split("-")[-1] in current_system.rules_text():
+                new_store.append([d])
+            elif d.startswith("-") and d.split("-")[-1] in previous_system.rules_text():
+                prev_store.append([d])
```

Comment:
This chunk of code looks very similar to the code in `ConfirmDeploymentDialog`. Can it be abstracted out and reused? I'd suggest making it a separate method in the file, but outside of the class so it can be imported into `ConfirmDeploymentDialog` and reused.